spdx-utils/src/main/kotlin/model/SpdxDocument.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.spdx.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonProperty
+
+private const val SPDX_VERSION = "SPDX-2.2"
+private const val DATA_LICENSE = "CC0-1.0"
+
+/**
+ * An SPDX document as specified by https://github.com/spdx/spdx-spec/tree/master/chapters and
+ * https://github.com/spdx/spdx-spec/blob/development/v2.2.1/examples/.
+ */
+data class SpdxDocument(
+    /**
+     * Identifier og this [SpdxDocument] which may be referenced in relationships by other files, packages internally
+     * and documents externally.
```

Comment:
The document contains a list of packages which can refer to this document creating a cyclic reference.

Further this document may be reference by other documents also via this ID, so that reference is external to the document.

Reply:
Thanks.